src/libraries/System.Net.Quic/src/System/Net/Quic/Interop/msquic.cs

```diff
@@ -108,72 +110,19 @@ public NativeTypeNameAttribute(string name)
         public string Name => _name;
     }
 
-    [StructLayout(LayoutKind.Explicit)]
-    internal struct QuicAddrFamilyAndLen
-    {
-        [FieldOffset(0)]
-        public ushort sin_family;
-        [FieldOffset(0)]
-        public byte sin_len;
-        [FieldOffset(1)]
-        public byte sin_family_bsd;
-    }
-
-    [StructLayout(LayoutKind.Sequential)]
-    internal unsafe struct QuicAddrIn
-    {
-        public QuicAddrFamilyAndLen sin_family;
-        public ushort sin_port;
-        public fixed byte sin_addr[4];
-    }
-
-    [StructLayout(LayoutKind.Sequential)]
-    internal unsafe struct QuicAddrIn6
-    {
-        public QuicAddrFamilyAndLen sin6_family;
-        public ushort sin6_port;
-        public uint sin6_flowinfo;
-        public fixed byte sin6_addr[16];
-        public uint sin6_scope_id;
-    }
-
-    [StructLayout(LayoutKind.Explicit)]
     internal struct QuicAddr
     {
-        [FieldOffset(0)]
-        public QuicAddrIn Ipv4;
-        [FieldOffset(0)]
-        public QuicAddrIn6 Ipv6;
-        [FieldOffset(0)]
-        public QuicAddrFamilyAndLen FamilyLen;
+        public SockAddr Address;
 
-        public static bool SockaddrHasLength => OperatingSystem.IsFreeBSD() || OperatingSystem.IsIOS() || OperatingSystem.IsMacOS() || OperatingSystem.IsMacCatalyst() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS();
+        public int Family => Address.Family;
+        public Interop.Sys.sockaddr_in6 Ipv6 => Address.Ipv6;
```

Comment:
This file is a verbatim copy of https://github.com/microsoft/msquic/blob/main/src/cs/lib/msquic.cs from msquic repo. I do not think you want to make custom edits to it that make it depend on the dotnet runtime details. It will future make synchronization of this file harder.

Reply:
The native QUIC_ADDR (e.g. source of truth) use `sockaddr`

https://github.com/microsoft/msquic/blob/cb9e250e13d9354ac0b325509ea2d7003294f3c6/src/inc/msquic_posix.h#L157-L161

Our sockets needs more than Quic and I did not want to drag the runtime check for `SockaddrHasLength`. Instead the proposed `struct` represents the OS layout. Problem with MsQuic is that it does not really have PAL at c# layer -> just split between Windows and POSIX. 

We _could_ drag the MsQuic definitions but use serialization from `Sockets`/`sockaddr` to produce the bits.

cc: @nibanks for additional thoughts.